redfish-core/lib/power_supply.hpp

```diff
@@ -248,13 +256,37 @@ inline void
             return;
         }
 
-        if (!value)
+        if (!value || !available)
         {
             asyncResp->res.jsonValue["Status"]["Health"] = "Critical";
         }
     });
 }
 
+inline void getPowerSupplyStateAndHealth(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& service, const std::string& path)
+{
+    sdbusplus::asio::getProperty<bool>(
+        *crow::connections::systemBus, service, path,
+        "xyz.openbmc_project.State.Decorator.Availability", "Available",
+        [asyncResp, service, path](const boost::system::error_code ec,
+                                   const bool available) {
+        if (ec)
+        {
+            if (ec.value() != EBADR)
+            {
+                messages::internalError(asyncResp->res);
```

Comment:
If you want to add logging, it might be helpful one day 
                BMCWEB_LOG_ERROR("DBUS response error {}",
                                 ec.value());

Reply:
done